### PR TITLE
Added an ability to track user id

### PR DESCRIPTION
**Pull request by viteksafronov**
This commit enables GA functionality for user id tracking
More info here https://support.google.com/analytics/answer/3123662

### CODE REVIEW

File: event/listener.php [hunk 1]
```diff
@@ -66,6 +66,10 @@ static public function getSubscribedEvents()
 	public function load_google_analytics()
 	{
 		$this->template->assign_var('GOOGLEANALYTICS_ID', $this->config['googleanalytics_id']);
+		if ($this->config['googleanalytics_track_user_id'] && $this->user->data['is_registered'])
+		{
+			$this->template->assign_var('GOOGLEANALYTICS_USER_ID', "ga('set', 'userId', {$this->user->data['user_id']});");
+		}
```
Comment (VSEphpbb) [inline event/listener.php]: This shouldn't call assign_var twice. It should instead use assign_vars for both these and set a conditional. And the template code should go in the template. Something more like:

```
$this->template->assign_vars(array(
	'GOOGLEANALYTICS_ID'       => $this->config['googleanalytics_id'],
	'GOOGLEANALYTICS_USER_ID'  => $this->user->data['user_id'],
	'S_GOOGLEANALYTICS_USER_ID => ($this->config['googleanalytics_track_user_id'] && $this->user->data['is_registered']),
));


<!-- IF S_GOOGLEANALYTICS_USER_ID -->
	ga('set', 'userId', {GOOGLEANALYTICS_USER_ID});
<!-- ENDIF -->
```

File: language/en/googleanalytics_acp.php [hunk 2]
```diff
@@ -38,7 +38,9 @@
...
 
 $lang = array_merge($lang, array(
-	'ACP_GOOGLEANALYTICS_ID'			=> 'Google Analytics ID',
-	'ACP_GOOGLEANALYTICS_ID_EXPLAIN'	=> 'Enter your Google Analytics ID code, e.g.: <samp>UA-0000000-00</samp>.',
-	'ACP_GOOGLEANALYTICS_ID_INVALID'	=> '“%s” is not a valid Google Analytics ID code.<br />It should be in the form “UA-0000000-00”.',
+	'ACP_GOOGLEANALYTICS_ID'					=> 'Google Analytics ID',
+	'ACP_GOOGLEANALYTICS_ID_EXPLAIN'			=> 'Enter your Google Analytics ID code, e.g.: <samp>UA-0000000-00</samp>.',
+	'ACP_GOOGLEANALYTICS_ID_INVALID'			=> '“%s” is not a valid Google Analytics ID code.<br />It should be in the form “UA-0000000-00”.',
+	'ACP_GOOGLEANALYTICS_TRACK_USER_ID'			=> 'Track user id',
+	'ACP_GOOGLEANALYTICS_TRACK_USER_ID_EXPLAIN'	=> 'If enabled, Google Analytics will track the unique userId value for each authenticated user.',
```
Comment (VSEphpbb) [inline language/en/googleanalytics_acp.php]: Needs to mention that this also needs enabling, setting up and agreeing to policies over at google first.

By the way, this whole PR is unnecessary if you just use the Google tag Manager, and simply give it the name of the phpBB cookie that ends in _uid. https://support.google.com/tagmanager/answer/4565987

File: language/ru/googleanalytics_acp.php [hunk 3]
```diff
@@ -0,0 +1,46 @@
+<?php
```
Comment (michaelcullum) [inline language/ru/googleanalytics_acp.php]: This shouldn't be here

Comment (viteksafronov) [inline language/ru/googleanalytics_acp.php]: You mean this opening tag or the whole file? Looks like it was committed by mistake and should be added via another PR.

Comment (michaelcullum) [inline language/ru/googleanalytics_acp.php]: The file yeah.

**Comment by VSEphpbb** [comment]
Replaced by #63
